File: P4/Civitas/src/civitas/CasillaSorpresa.java
package civitas;

import java.util.ArrayList;

/**
 *
 * @author salva
 */
public class CasillaSorpresa extends Casilla{
    private MazoSorpresas mazo;
    //private Sorpresa sorpresa;
    
    CasillaSorpresa(MazoSorpresas mazo, String nombre){
        super(nombre);
        this.mazo = mazo;
    //    this.sorpresa = null;
    }
    
    @Override
    public String toString(){
//        if(sorpresa == null){
//            sorpresa = mazo.siguiente();
//        }
//        String info = super.toString();
//        info += "\n¡Has caído en una sorpresa!";
//        info += sorpresa.toString();
//        return info;
        String s = super.toString();
        return s;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos)) {
            Sorpresa sorpresa = mazo.siguiente();
            informe(actual, todos);
            sorpresa.aplicarAJugador(actual, todos);
        }
    }
}

File: P4/civitasruby/lib/civitas/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas

  class Casilla_sorpresa < Casilla
    
    def initialize(mazo, nombre) # deberiamos pasarle nombre como parametro?
      @mazo = mazo
      @sorpesa = mazo.siguiente
      super(nombre) 
    end
    
    @override
    def to_s
      "#{super}   La proxima sorpresa es: #{@sorpresa.to_s}"
    end
    
    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        @sorpresa = @mazo.siguiente
        super
        @sorpresa.aplicar_a_jugador(actual, todos)
      end
    end
    
  end

end
